infinispan/runtime/src/main/java/io/quarkiverse/langchain4j/infinispan/runtime/InfinispanEmbeddingStoreConfig.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkiverse.langchain4j.infinispan.runtime;
+
+import static io.quarkus.runtime.annotations.ConfigPhase.RUN_TIME;
+
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefault;
+
+/**
+ * Configuration of the Redis embedding store.
+ */
+@ConfigRoot(phase = RUN_TIME)
+@ConfigMapping(prefix = "quarkus.langchain4j.infinispan")
+public interface InfinispanEmbeddingStoreConfig {
+
+    /**
+     * The dimension of the embedding vectors. This has to be the same as the dimension of vectors produced by
+     * the embedding model that you use. For example, AllMiniLmL6V2QuantizedEmbeddingModel produces vectors of dimension 384.
+     * OpenAI's text-embedding-ada-002 produces vectors of dimension 1536.
+     */
+    Long dimension();
```

Comment:
Open question: do we really need this? 

I found it super confusing when using Redis.

Reply:
Yes, the size of the vector is needed to define the index, and depending on the type of data/vector of your embeddings, the size can be diff. This is configurable in every embedding store in langchain

Reply:
But please make it `Optional<Long>` so it's not required in cases when the schema already exists. If then, during boot, you find out that the cache needs to be created, and the dimension wasn't specified, then you throw an error.

Reply:
ok, let's do that

Reply:
the dimension is also used to know which index type is needed. it's a but more complex than that, I need to grab the cache configuration and infer the indexed entity name.
I decided the name for this use case, that could be parametrable

Reply:
ok, I can't do this easily. Grabbing the configuration of a cache, for security reasons, is not an available method from the client. This action is only for "admins" and the user of this api might not have those rights.
What I can do is either specifying the vector size, either the indexed entity name

Reply:
ok, right now dimension is mandatory. Schemas need to exist in the server and the client. I can't make it "optional" for this use case.

Reply:
IMO, at least for now, schemas for vectors need to follow an specific convection in Infinispan. Caches can exist or not. The dimension is on the schema.

Reply:
Ah, so you're saying that the client requires to have its own copy of the schema definition (and it can't ask the server to provide it for an existing cache) and that's why the client always needs to have the dimension configured?
What happens if the cache exists on the server, but the client has a WRONG dimension configured?

Reply:
The serialization is done with protostream (protobuf) and hibernate search.
If the cache exists with LangchainItem384 (schema on client and server)
But then I try the existing cache with items on 384 with another dimension (383)
The server will upload a new schema for that size, register on the client the serialization context for that dimension and then Hibernate Search will say: 

```org.infinispan.client.hotrod.exceptions.HotRodClientException:Request for messageId=15 returned server error (status=0x85): org.hibernate.search.util.common.SearchException: ISPN014505: Unknown entity name: 'LangchainItem383'.```

Reply:
this is mandatory in the cache config for this use case
```json
"indexing": {
        "enabled": true,
        "storage": "local-heap",
        "indexed-entities": [
          "LangchainItem384"
        ]
      },
```
They need to match

Reply:
Ok thanks for explaining, then it makes sense to me

Reply:
We can improve the error message to be more clear on this particular integration, however things can evolve and get even better from our side. for 15 a bunch of things are coming like protobuf 3 integration and some API improvements and capabilities. The code I did here, will still work, but will be improved